File: workflow_shell/git/pull.py
from ..util import get_result
from ..util import run
from ..cli import all_commands

command_string = 'gp'
command_help = 'Takes one optional argument, ' + \
    'the name of the base branch to pull and rebase with'
command_arguments = ['base_branch']
argument_required = False
argument_default = 'develop'


def main(*args, **kwargs):
    context = args[0]
    original_branch = get_result(['git', 'rev-parse', '--abbrev-ref', 'HEAD'])

    # Check out and pull changes for the base branch
    base_branch = kwargs.get(command_arguments[0])
    context.invoke(all_commands['gcam'], base_branch=base_branch)

    # Go back to original branch and rebase
    run(['git', 'checkout', original_branch])
    run(['git', 'rebase', base_branch])

File: src/commands/git/Pull.ts
import { Option } from '../../common/interface/Option';
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Argument } from '../../common/interface/Argument';
import { Command as GitSyncBase } from './SyncBase';
import { Command as GitRebase } from './Rebase';
import { Names } from '../../common/interface/Names';
import { getCurrentBranchName } from '../../common/Git';

const argumentss: Argument[] = [Argument.BASE_BRANCH];
const options: Option[] = [];

export class Command extends SuperCommand {
  description: string = 'Takes one optional argument, the name of the base branch to pull and rebase with';
  alias: string = 'gp';
  public async execute(vital?: boolean, input?: any) {
    await super.executeWithInput(argumentss, options, input, vital, async () => {
      const originalBranch = await getCurrentBranchName();
      if (this.input[Names.NUMBER]) await (new GitRebase()).execute(true, this.input);
      await (new GitSyncBase()).execute(true, this.input);
      await executeCommand('git', ['checkout', originalBranch]);
      await executeCommand('git', ['rebase', this.input[Names.BASE_BRANCH]]);
    });
  }
}
